common/src/keepassPlugin/databaseInterface/private/Keepass2DatabaseInterface.cpp

```diff
@@ -96,23 +100,27 @@ void Keepass2DatabaseInterface::slot_openDatabase(QString filePath, QString pass
     }
 
     CompositeKey masterKey;
-    masterKey.addKey(PasswordKey(password));
+    auto passwordKey = QSharedPointer<PasswordKey>::create();
+    passwordKey->setPassword(password);
+    masterKey.addKey(passwordKey);
     if (!keyfile.isEmpty()) {
-        FileKey key;
+        auto key = QSharedPointer<FileKey>::create();
         QString errorMsg;
-        if (!key.load(keyfile, &errorMsg)) {
+        if (!key->load(keyfile, &errorMsg)) {
             emit databaseOpened(DatabaseAccessResult::RE_KEYFILE_OPEN_ERROR, errorMsg);
             return;
         }
         masterKey.addKey(key);
     }
 
     if (m_Database) {
-        delete m_Database;
+        m_Database.clear();
+        //delete m_Database;
     }
 
-    KeePass2Reader reader;
-    m_Database = reader.readDatabase(&file, masterKey);
+    KdbxXmlReader reader(KeePass2::FILE_VERSION_4);
```

Comment:
I think this will be a problem because here the database reader is set hard to database version 4. That means that it will not open Keepass 2 database version 2, 3 and 3.1 any more. IIRC there is a function to check for the file type in the keepassxc 2 database sources. We should adapt it here and use the correct reader depending on the keepass 2 database file version.

Reply:
-> KeePass2Reader::readDatabase(const QString& filename, QSharedPointer<const CompositeKey> key, Database* db)